### PR TITLE
Reduce compile dependencies for API/SDK.

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

**Comment by jkwatson** [comment]
this looks like a good, simple change to me. 👍

File: opentracing_shim/src/test/java/io/opentelemetry/opentracingshim/testbed/TestUtils.java [hunk 1]
```diff
@@ -67,7 +67,7 @@ public Integer call() throws Exception {
         new Condition() {
           @Override
           public boolean check(Span span) {
-            AttributeValue attrValue = span.getAttributes().getAttributeMap().get(key);
+            AttributeValue attrValue = span.getAttributes().getAttributeMapMap().get(key);
```
Comment (jkwatson) [inline opentracing_shim/src/test/java/io/opentelemetry/opentracingshim/testbed/TestUtils.java]: did bumping the autovalue version make this method change?

Comment (bogdandrutu) [reply opentracing_shim/src/test/java/io/opentelemetry/opentracingshim/testbed/TestUtils.java]: No, the `options.compilerArgs += ["-Xlint:-deprecation"]` I probably should do that separately.

File: build.gradle [hunk 2]
```diff
@@ -150,10 +148,12 @@ subprojects {
     }
 
     dependencies {
-        api libraries.auto_value_annotation,
```
Comment (adriancole) [inline build.gradle]: I hate to mention this, but where used and used in android, the annotation jar needs to be set to api. This is one reason we use our own annotations in brave core jar..

https://github.com/google/auto/blob/master/value/userguide/index.md#in-pomxml

Comment (bogdandrutu) [reply build.gradle]: Do you know why?

Comment (adriancole) [reply build.gradle]: no I just noticed some android people mentioning this.. maybe @JakeWharton remembers?

Comment (bogdandrutu) [reply build.gradle]: @adriancole if you hear again about this and you will hear the reason please let me know.

Comment (adriancole) [reply build.gradle]: well the project is from one of your colleagues @eamonnmcmanus so no need be to be a proxy :p

anyway here's an issue I fished from searching the repo https://github.com/google/auto/issues/620

File: opentracing_shim/src/test/java/io/opentelemetry/opentracingshim/testbed/TestUtils.java (hunk 1 above)
Comment (bogdandrutu) [reply opentracing_shim/src/test/java/io/opentelemetry/opentracingshim/testbed/TestUtils.java]: Done in a different PR.

**Pull request merged**